wazo_phoned/plugin_helpers/client/http.py

```diff
@@ -0,0 +1,247 @@
+# Copyright 2019 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import logging
+import re
+
+from collections import namedtuple
+from flask import (
+    render_template,
+    request,
+    Response,
+)
+from operator import attrgetter
+from requests.exceptions import ConnectionError, HTTPError
+
+from wazo_phoned.auth_remote_addr import AuthResource
+
+from .exceptions import WazoAuthConnectionError, WazoDirdConnectionError, NoSuchUser
+from .schema import UserUUIDSchema, LookupSchema
+from ..common import output_error
+
+
+logger = logging.getLogger(__name__)
+
+_PhoneFormattedResult = namedtuple('_PhoneFormattedResult', ['name', 'number'])
+
+
+class ClientMenu(AuthResource):
+
+    content_type = None
+    template = None
+
+    def __init__(self, vendor, auth_client, dird_client):
+        self.vendor = vendor
+        self.auth_client = auth_client
+        self.dird_client = dird_client
+        super().__init__()
+
+    def get(self, profile):
+        args = UserUUIDSchema().load(request.args)
+        xivo_user_uuid = args['xivo_user_uuid']
+
+        response_rendered = render_template(
+            self.template,
+            xivo_user_uuid=xivo_user_uuid,
+            xivo_proxy_url=_build_next_url('menu'),
+        )
+
+        return Response(response_rendered, content_type=self.content_type, status=200)
+
+
+class ClientInput(AuthResource):
+
+    content_type = None
+    template = None
+
+    def __init__(self, vendor, auth_client, dird_client):
+        self.vendor = vendor
+        self.auth_client = auth_client
+        self.dird_client = dird_client
+        super().__init__()
+
+    def get(self, profile):
+        args = UserUUIDSchema().load(request.args)
+        xivo_user_uuid = args['xivo_user_uuid']
+
+        response_rendered = render_template(
+            self.template,
+            xivo_user_uuid=xivo_user_uuid,
+            xivo_proxy_url=_build_next_url('input'),
+        )
+
+        return Response(response_rendered, content_type=self.content_type, status=200)
+
+
+class ClientLookup(AuthResource):
+
+    MAX_ITEM_PER_PAGE = None
+    content_type = None
+    template = None
+
+    def __init__(self, vendor, auth_client, dird_client):
+        self.vendor = vendor
+        self.auth_client = auth_client
+        self.dird_client = dird_client
+        super().__init__()
+
+    def get(self, profile):
+        args = LookupSchema().load(request.args)
+        limit = args['limit'] if args['limit'] is not None else self.MAX_ITEM_PER_PAGE
```

Comment:
If I understand well, this option is only to facilitate test?

Reply:
Are you talking about MAX_ITEM_PER_PAGE? If so, that is because some phones cannot display more than a certain amount of items per page (Aastra is 16 max items per page for example).

Reply:
No I talk about be able to set the limit in the API. I mean, no telephone will send limit field

Reply:
Ok. Indeed it is mostly useful for tests!